File: src/dataset/typescript/difference-of-squares_exercism-samajammin/gemini/GeminiFlash_difference-of-squares_v3.ts
export default class Squares {
  squareOfSum: number;
  sumOfSquares: number;
  difference: number;

  constructor(num: number) {
    this.squareOfSum = this.calcSquareOfSum(num);
    this.sumOfSquares = this.calcSumOfSquares(num);
    this.difference = this.squareOfSum - this.sumOfSquares;
  }

  private calcSquareOfSum(num: number): number {
    const sum = (num * (num + 1)) / 2;
    return sum * sum;
  }

  private calcSumOfSquares(num: number): number {
    return (num * (num + 1) * (2 * num + 1)) / 6;
  }

  createFirstN(num: number): number[] {
    const result: number[] = [];
    for (let i: number = 1; i <= num; i++) {
      result.push(i);
    }
    return result;
  }
}

File: src/dataset/cpp/difference-of-squares_exercism-pawelo091991/gemini/GeminiFlash_difference_of_squares_v3.cpp
#include "difference_of_squares.h"
#include <numeric>

int squares::square_of_sum(int number) {
    long long sum = static_cast<long long>(number) * (number + 1) / 2;
    return sum * sum;
}

int squares::sum_of_squares(int number) {
    return number * (number + 1) * (2 * number + 1) / 6;
}

int squares::difference(int number) {
    return square_of_sum(number) - sum_of_squares(number);
}